utils/src/main/java/io/github/mzmine/util/files/FileAndPathUtil.java

```diff
@@ -690,6 +703,72 @@ public static File createTempFile(String prefix, String suffix) throws IOExcepti
     return tempFile;
   }
 
+  private static final SecureRandom random = new SecureRandom();
+
+  private static Path generateRandomPath(String prefix, String suffix, Path dir) {
+    String s = prefix + Long.toUnsignedString(random.nextLong()) + suffix;
```

Comment:
is next long thread safe?

Reply:
does not have to be thread safe. The filename will be checked while creating the file and this is an atomic operation on the filesystem

Reply:
if FileChannel.open fails it will try another filename